diff -up ROCR-Runtime-rocm-5.7.1/src/core/runtime/trap_handler/CMakeLists.txt.omv~ ROCR-Runtime-rocm-5.7.1/src/core/runtime/trap_handler/CMakeLists.txt
--- ROCR-Runtime-rocm-5.7.1/src/core/runtime/trap_handler/CMakeLists.txt.omv~	2023-12-02 23:06:42.107323090 +0100
+++ ROCR-Runtime-rocm-5.7.1/src/core/runtime/trap_handler/CMakeLists.txt	2023-12-02 23:07:06.727584218 +0100
@@ -65,7 +65,7 @@ function(gen_kernel_bc TARGET_ID INPUT_F
   set(CODE_OBJECT "${OUTPUT_FILE}.hsaco")
 
   separate_arguments(CLANG_ARG_LIST UNIX_COMMAND
-  "-x assembler -target amdgcn-amd-amdhsa -mcpu=${TARGET_ID} -o ${CODE_OBJECT} ${INPUT_FILE}")
+  "-x assembler -target amdgcn-amd-amdhsa -mcpu=${TARGET_ID} -Wl,--no-gc-sections -o ${CODE_OBJECT} ${INPUT_FILE}")
 
   ## Add custom command to produce a code object file.
   add_custom_command(OUTPUT ${CODE_OBJECT} COMMAND clang ${CLANG_ARG_LIST}
