diff -up ROCR-Runtime-rocm-5.7.1/src/CMakeLists.txt.omv~ ROCR-Runtime-rocm-5.7.1/src/CMakeLists.txt
--- ROCR-Runtime-rocm-5.7.1/src/CMakeLists.txt.omv~	2023-12-02 23:11:16.710229777 +0100
+++ ROCR-Runtime-rocm-5.7.1/src/CMakeLists.txt	2023-12-02 23:12:51.823231319 +0100
@@ -273,7 +273,7 @@ if(${IMAGE_SUPPORT})
 endif()
 
 ## Link dependencies.
-target_link_libraries ( ${CORE_RUNTIME_TARGET} PRIVATE hsakmt::hsakmt PkgConfig::drm)
+target_link_libraries ( ${CORE_RUNTIME_TARGET} PRIVATE hsakmt::hsakmt PkgConfig::drm drm_amdgpu)
 target_link_libraries ( ${CORE_RUNTIME_TARGET} PRIVATE elf::elf dl pthread rt )
 
 ## Set the VERSION and SOVERSION values
